webpack.config.js

```diff
@@ -0,0 +1,64 @@
+const path = require("path");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const ReactRefreshWebpackPlugin = require("@pmmmwh/react-refresh-webpack-plugin");
+
+let mode = "development";
+if (process.env.NODE_ENV === "production") {
+  mode = "production";
+}
+
+const plugins = [
+  new MiniCssExtractPlugin({
+    filename: "[name].css",
+  }),
+];
+
+if (process.env.SERVE) {
+  plugins.push(new ReactRefreshWebpackPlugin());
+}
+
+module.exports = {
+  mode,
+  plugins,
+  entry: "./src/index.js",
+  devtool: "source-map",
+  output: {
+    filename: "[name].js",
+    path: path.resolve(__dirname, "./dist"),
+    publicPath: "/dist",
+    assetModuleFilename: "assets/[hash][ext][query]",
+  },
+  devServer: {
+    hot: true,
+  },
+  module: {
+    rules: [
+      {
+        test: /\.jsx?$/,
```

Comment:
```suggestion
        test /\.(js|jsx)$/,
```
We can write some helpers and other files)